linkerd/app/inbound/src/require_identity.rs

```diff
@@ -0,0 +1,73 @@
+use crate::target::TcpAccept;
+use indexmap::IndexSet;
+use linkerd_app_core::{svc::stack::Predicate, Conditional, Error};
+use std::sync::Arc;
+
+/// A connection policy that fails direct connections that don't have a client
+/// identity.
+#[derive(Clone, Debug)]
+pub struct RequireIdentityForDirect;
+
+/// A connection policy that fails connections that don't have a client identity
+/// if they target one of the configured local ports.
+#[derive(Clone, Debug)]
+pub struct RequireIdentityForPorts {
+    ports: Arc<IndexSet<u16>>,
+}
+
+#[derive(Debug)]
+pub struct IdentityRequired(());
+
+// === impl RequireIdentityForDirect ===
+
+impl Predicate<TcpAccept> for RequireIdentityForDirect {
+    type Request = TcpAccept;
+
+    fn check(&mut self, meta: TcpAccept) -> Result<TcpAccept, Error> {
+        tracing::debug!(client.id = ?meta.client_id);
+        match meta.client_id {
+            Conditional::Some(Some(_)) => {}
+            _ => return Err(IdentityRequired(()).into()),
+        }
+        Ok(meta)
```

Comment:
nit/tioli: isn't this just 
```suggestion
        match meta.client_id {
            Conditional::Some(Some(_)) => Ok(meta),
            _ =>Err(IdentityRequired(()).into(),
        }
```
or does the borrow checker not like that?